packages/contracts-ops/contracts/CallBatch.sol

```diff
@@ -57,55 +97,111 @@ abstract contract CallBatch is Script {
         buffer.writeObjectClose(indent, terminal);
     }
 
-    function writeCallList(JsonWriter.Buffer memory buffer) private {
-        string memory indent = "";
-        string memory inner = indent.nextIndent();
-        string memory innerer = inner.nextIndent();
-        buffer.writeLine(indent, "{");
-        buffer.writeArrayOpen(inner, domain);
-        for (uint32 i = 0; i < calls.length; i++) {
-            writeCall(buffer, innerer, calls[i], i == calls.length - 1);
+    function writeCallList(JsonWriter.Buffer memory buffer, bool local)
+        private
+    {
+        if (local) {
+            string memory indent = "";
+            string memory inner = indent.nextIndent();
+            string memory innerer = inner.nextIndent();
+            buffer.writeLine(indent, "{");
+            buffer.writeArrayOpen(inner, localDomain);
+            for (uint32 i = 0; i < localCalls.length; i++) {
+                writeCall(
+                    buffer,
+                    innerer,
+                    localCalls[i],
+                    i == localCalls.length - 1
+                );
+            }
+            buffer.writeArrayClose(inner, true);
+            buffer.writeLine(indent, "}");
+        } else {
+            for (uint256 j; j < remoteDomains.length; j++) {
+                string memory indent = "";
+                string memory inner = indent.nextIndent();
+                string memory innerer = inner.nextIndent();
+                buffer.writeLine(indent, "{");
+                buffer.writeArrayOpen(inner, remoteDomains[j]);
+                for (uint32 i = 0; i < remoteCalls[j].length; i++) {
+                    writeCall(
+                        buffer,
+                        innerer,
+                        remoteCalls[j][i],
+                        i == remoteCalls[j].length - 1
+                    );
+                }
+                buffer.writeArrayClose(inner, true);
+                buffer.writeLine(indent, "}");
+            }
         }
-        buffer.writeArrayClose(inner, true);
-        buffer.writeLine(indent, "}");
     }
 
     function finish() public {
-        require(bytes(domain).length != 0, "must initialize");
-        require(bytes(outputFile.path).length != 0, "must initialize");
+        require(bytes(localDomain).length != 0, "must initialize");
+        require(bytes(outputFileLocal.path).length != 0, "must initialize");
         require(!complete, "already written");
         complete = true;
         JsonWriter.Buffer memory buffer = JsonWriter.newBuffer();
-        writeCallList(buffer);
-        buffer.flushTo(outputFile);
+        writeCallList(buffer, true);
```

Comment:
should we flush each domain's governance calls to individual files?

Reply:
```
./actions/governance-ethereum.json
./actions/governance-avalanche.json
```
etc

Reply:
Depending on what we want. If we use vm.writeJson, we can create arbitrarily complex JSON structures quite easily. With the json library we have, we can't create anything nested, so the best we can do is flush them to different files.

Reply:
i think we definitely need to do that actually. as things stand i think the remote file wouldn't be actual JSON. and i don't think it would be clear which governance actions are for which domains

Reply:
I think it should be 1 run, 1 output file. creating an unknown number or network-dependent number seems gross